### ISSUE TITLE
Unavailable log types shown when logging

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#11413]
### Describe your problem!

For some reason c:geo shows more log types than actually are available.

Web:
![grafik](https://user-images.githubusercontent.com/949669/128330775-c0c63009-6d14-4ae0-8825-e2eb7be9c153.png)

c:geo:
![grafik](https://user-images.githubusercontent.com/949669/128330898-f06108cf-9582-4aa9-a4ac-170be13d64eb.png)

The NM and NA log types should not be shown.
If you try to send a NM and NA log the log sending fails anyway,

### How to reproduce?

- Start logging some cache (not your owned or found)
- Tap on log type to see available log types

### Actual result after these steps?

It shows:
- Found
- Did not find
- Write note
- Needs maintenance
- Needs archived

### Expected result after these steps?

It should show:
- Found
- Did not find
- Write note

NM and NA is achieved via "Report problem"

### Reproducible

Yes

### c:geo Version

2021.07.30-RC

### Additional Information

I am not sure why this happens. We should have a code part which checks the available log types when the logging page is loaded.
This seems to work as e.g. for my owned caches I do see the correct log types.

I also checked 
Webcam caches = Same problem (NM and NA shown)
Event caches = Same problem (NA shown)

**Comment by bekuno** [comment]
This is an (old) implemented behavior, as I learned in #11226.

**Comment by fm-sys** [comment]
I just used this feature to archive caches via c:geo some weeks ago and it _did_ work! 

Or to be more precise: The first try did fail, but after doing the same thing, just resubmitting the log via c:geo, the log was posted successfully. 

While this log type is not shown in the UI of their website, the webAPI we are using does still support these types...

**Comment by Lineflyer** [comment]
> I just used this feature to archive caches via c:geo some weeks ago and it did work!

That is for your owned caches. Of course this works as for your owned caches the log types "Archive" and "Temporarily Disable" (etc.) are available.

We are here talking about reporting a needs maintenance/archived for not-owned caches

> While this log type is not shown in the UI of their website, the webAPI we are using does still support these types...

No.

**Comment by Lineflyer** [comment]
For the avoidance of doubt (tested this many times while the new log page with WebAPI was implemented into c:geo):

The log types shown on the webpage (new logging page) have to be identical to those shown in c:geo!
Only available log types can be posted via the WebAPI.

Normally there are: Found, DNF, Write Note
For owned and found caches (and for some special caches) this differs.

We have a piece of code which is ensuring this but for some reason now more log types are shown, but those not shown on the webpage will not work to be posted by c:geo (just tested it).

I will use the preUI release to doublecheck the behavior with that, but I am quite sure it will be working as expected there..lets see.

**Comment by Lineflyer** [comment]
> This is an (old) implemented behavior, as I learned in #11226.

No, back in the time when c:geo used the old logging page (from server), we were able to direct post a NM or NA log. Nowadays as we use the WebAPI (new logging page) it has to be done with a DNF, Note and associated NM,NA message.

This leads to an inconsistency when using offline logs (where it is still possible). This is the topic of #11226.
When logging online no NM and NA are possible as direct logging types...did this maybe come in with the change related to #11266? Doesn't look like that as it only deal with visualization of offline logs.

**Comment by bekuno** [comment]
I meant with "old" our last release.

I think (can not test) that for a NM log a Note log with NM attribute should be generated.

**Comment by bekuno** [comment]
See also the initial problem in https://github.com/cgeo/cgeo/issues/11226#issuecomment-879796245 , where two logs with the same timestamp are generated in c:geo and therefore the ordering PR was created.

**Comment by Lineflyer** [comment]
I just tested with the `preUI_release` and those NM and NA log types show up and work. Now I am a bit puzzled and have to apologize..need to test again with new release why it does no longer work there.

**Comment by Lineflyer** [comment]
It seems I do now know whats going on  (not verified yet, but sounds logical):

There seems to be a restriction on server side, that NM and NA can only be logged with todays date (server time?) but not in the past.
This in combination with the fact, that we have that date picker problem might lead to the error messages received on support, that NA and NM cannot be logged.

If I try to log a NM with a date some weeks ago, I can reproduce the failure to send the log.

**Comment by Lineflyer** [comment]
So, if I got it right, logging NM and NA will work but only if logging date=today.
I will rename the issue accordingly, and the severity is not so high anymore.
